src/incubator/TouchableOpacity2.tsx

```diff
@@ -0,0 +1,169 @@
+import {asBaseComponent, forwardRef, BaseComponentInjectedProps, ForwardRefInjectedProps} from '../commons/new';
+import {ViewProps} from '../components/view';
+
+export type TouchableOpacityProps = {
+  /**
+   * Background color
+   */
+  backgroundColor?: string;
+  /**
+   * Background color when actively pressing the touchable
+   */
+  feedbackColor?: string;
+  /**
+   * Opacity value when actively pressing the touchable
+   */
+  activeOpacity?: number;
+  /**
+   * Scale value when actively pressing the touchable
+   */
+  activeScale?: number;
+  /**
+   * Callback for when tapping the touchable
+   */
+  onPress?: (props: any) => void;
+  /**
+   * Callback for when long pressing the touchable
+   */
+  onLongPress?: (props: any) => void;
+  /**
+   * Pass controlled pressState to track gesture state changes
+   */
+  pressState?: State;
+  /**
+   * If true, disable all interactions for this component.
+   */
+  disabled?: boolean;
+  /**
+   * Pass custom style
+   */
+  style?: ViewProps['style'];
+  onLayout?: (event: LayoutChangeEvent) => void;
+  testID?: string;
+};
+
+type Props = PropsWithChildren<TouchableOpacityProps & BaseComponentInjectedProps & ForwardRefInjectedProps>;
+
+/**
+ * @description: a Better, enhanced TouchableOpacity component
+ * @modifiers: flex, margin, padding, background
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/incubatorScreens/TouchableOpacityScreen.js
+ */
+function TouchableOpacity(props: Props) {
+  const {
+    children,
+    modifiers,
+    style,
+    // onPress = _.noop,
+    // onLongPress,
+    disabled,
+    forwardedRef,
+    feedbackColor,
+    activeOpacity = 0.2,
+    activeScale = 1,
+    ...others
+  } = props;
+  const {borderRadius, paddings, margins, alignments, flexStyle} = modifiers;
+
+  const isActive = useSharedValue(0);
+  /* This flag is for fixing an issue with long press triggering twice
+  TODO: Consider revisiting this issue to see if it still occurs */
+  const isLongPressed = useSharedValue(false);
+
+  const backgroundColor = useMemo(() => {
+    return props.backgroundColor || modifiers.backgroundColor;
+  }, [props.backgroundColor, modifiers.backgroundColor]);
+
+  const onPress = useCallback(() => {
+    props.onPress?.(props);
+  }, [props]);
+
+  const onLongPress = useCallback(() => {
+    props.onLongPress?.(props);
+  }, [props]);
+
+  const tapGestureHandler = useAnimatedGestureHandler({
+    onStart: () => {
+      isActive.value = withTiming(1, {duration: 200});
+    },
+    onEnd: () => {
+      isActive.value = withTiming(0, {duration: 200});
+      runOnJS(onPress)();
+    },
+    onFail: () => {
+      isActive.value = withTiming(0, {duration: 200});
+    }
+  });
+
+  const longPressGestureHandler = useAnimatedGestureHandler({
+    onActive: () => {
+      if (!isLongPressed.value) {
+        isLongPressed.value = true;
+        isActive.value = withTiming(0, {duration: 200});
```

Comment:
What is the meaning of the duration here? I can't see any difference between `{duration: 0}` and `{duration: 2000}`

Reply:
There's definitely a difference (: 
Try again. Try refreshing the screen after making the change in the code. 
It dictates the time it takes for the activeness transition.
I did a little refactor here cause these lines repeat itself so pull the new code